src/frontend/src/components/authenticateBox.ts

```diff
@@ -256,27 +280,83 @@ export const authenticateBoxFlow = async <T, I>({
   }
 };
 
+// A type representing flow errors present in most flows
+type FlowError =
+  | AuthFail
+  | BadPin
+  | BadChallenge
+  | WebAuthnFailed
+  | UnknownUser
+  | ApiError
+  | RegisterNoSpace;
+
+// Type machinery to allow translating error code (e.g. kind: "authFail") using the actual data
+// contained in the error objects (e.g. authFail: (authFail: AuthFail) => ...).
+type ErrorKind = FlowError["kind"]; // all known error tags for FlowError
```

Comment:
Do they all the FlowError have a `"kind"` property?

Reply:
Yes! Otherwise this would be a type error I think